components/script/dom/document.rs

```diff
@@ -1735,7 +1735,7 @@ impl Document {
                 self.process_deferred_scripts();
             },
             LoadType::PageSource(_) => {
-                if self.has_browsing_context {
+                if self.has_browsing_context && self.is_fully_active() {
```

Comment:
Which part of the spec does this check come from?

Reply:
I think both are actually not spec compliant, yet necessary as bandaid because of the fact that layout will have already exited and we'll panic if we continue here if the document is not "fully active". 

I have filed https://github.com/servo/servo/issues/22507 to investigate the underlying issue...

What do you think?